chips/rp2040/src/pwm.rs

```diff
@@ -0,0 +1,1354 @@
+/// | 18    | 1A  |
+/// | 19    | 1B  |
+/// | 20    | 2A  |
+/// | 21    | 2B  |
+/// | 22    | 3A  |
+/// | 23    | 3B  |
+/// | 24    | 4A  |
+/// | 25    | 4B  |
+/// | 26    | 5A  |
+/// | 27    | 5B  |
+/// | 28    | 6A  |
+/// | 29    | 6B  |
+///
+/// **Note**:
+///
+/// + The same PWM output can be selected on two GPIO pins. The same signal will appear on each
+/// GPIO.
+/// + If a PWM B pin is used as an input, and is selected on multiple GPIO pins, then the PWM
+/// channel will see the logical OR of those two GPIO inputs
+impl From<RPGpio> for ChannelNumber {
+    fn from(gpio: RPGpio) -> Self {
+        match gpio as u8 >> 1 & 7 {
+            0 => ChannelNumber::Ch0,
+            1 => ChannelNumber::Ch1,
+            2 => ChannelNumber::Ch2,
+            3 => ChannelNumber::Ch3,
+            4 => ChannelNumber::Ch4,
+            5 => ChannelNumber::Ch5,
+            6 => ChannelNumber::Ch6,
+            7 => ChannelNumber::Ch7,
+            // This branch can't be reached due to logical AND
+            _ => panic!("Unreachable branch"),
+        }
+    }
+}
+
+/// Identifier for a channel pin
+///
+/// Each PWM channel has two pins: A and B.  
+/// Pin A is always configured as an output pin.  
+/// Pin B is configured as an output pin when running in free running mode. Otherwise, it is
+/// configured as an input pin.
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum ChannelPin {
+    A,
+    B,
+}
+
+/// Check ChannelNumber implementation for more details
+impl From<RPGpio> for ChannelPin {
+    fn from(gpio: RPGpio) -> Self {
+        match gpio as u8 & 1 {
+            0 => ChannelPin::A,
+            1 => ChannelPin::B,
+            // This branch can't be reached due to logical AND
+            _ => panic!("Unreachable branch"),
+        }
+    }
+}
+
+/// PWM channel configuration structure
+///
+/// This helper struct allows multiple channels to share the same configuration.
+pub struct PwmChannelConfiguration {
+    en: bool,
+    ph_correct: bool,
+    a_inv: bool,
+    b_inv: bool,
+    divmode: DivMode,
+    int: u8,
+    frac: u8,
+    cc_a: u16,
+    cc_b: u16,
+    top: u16,
+}
+
+impl PwmChannelConfiguration {
+    /// Create a set of default values to use for configuring a PWM channel:
+    /// + the channel is disabled
+    /// + trailing-edge modulation configured
+    /// + no pin A and B polarity inversion
+    /// + free running mode for the fractional clock divider
+    /// + integral part of the divider is 1 and the fractional part is 0
+    /// + compare values for both pins are set 0 (0% duty cycle)
+    /// + top value is set to its maximum value
+    pub fn default_config() -> Self {
+        PwmChannelConfiguration {
+            en: false,
+            ph_correct: false,
+            a_inv: false,
+            b_inv: false,
+            divmode: DivMode::FreeRunning,
+            int: 1,
+            frac: 0,
+            cc_a: 0,
+            cc_b: 0,
+            top: u16::MAX,
+        }
+    }
```

Comment:
What about implementing the `Default` trait?

Reply:
Implemented.